packages/utils/src/crypto/packet/padding.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { addPadding, removePadding, PADDING_TAG, PADDING_TAG_LENGTH } from './padding'
+import { u8aEquals } from '../../u8a'
+import assert from 'assert'
+import { PAYLOAD_SIZE } from './constants'
+
+describe('padding', function () {
+  it('test adding padding and removing it', function () {
+    const TEST_MESSAGE = new TextEncoder().encode('text')
+
+    const paddedMsg = addPadding(TEST_MESSAGE)
+
+    assert(paddedMsg.length == PAYLOAD_SIZE)
+
+    assert(u8aEquals(removePadding(paddedMsg), TEST_MESSAGE))
+  })
+
+  it('test adding padding and removing it - false positives', function () {
+    const TEST_MESSAGE = new TextEncoder().encode('text')
+
+    const paddedMsg = Uint8Array.from([
+      ...new Uint8Array(PAYLOAD_SIZE - TEST_MESSAGE.length - PADDING_TAG_LENGTH),
+      ...new TextEncoder().encode('H-PR'),
+      ...TEST_MESSAGE
+    ])
+
+    assert.throws(() => removePadding(paddedMsg), Error(`General error.`))
```

Comment:
need better error messages

Reply:
same here

Reply:
This is literally in our test suite - throwing a non-specific error is insanity.

Reply:
obsolete now, because code has changed